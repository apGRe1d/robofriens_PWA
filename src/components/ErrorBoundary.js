import React from 'react';

class ErrorBoundary extends React.Component {

  constructor() {
    super();
    this.state = {
      hasError: false
    }
  }

  // static getDerivedStateFromError(error) {
  //   return {hasError: true}
  // }

  componentDidCatch(error, errorInfo) {
    console.log({error});
    console.log({errorInfo});
    this.setState({hasError: true})
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong..</h1>
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
