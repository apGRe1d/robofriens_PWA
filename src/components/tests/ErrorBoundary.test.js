import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from '../ErrorBoundary';

it('when ErrorBoundary fires', function () {
  const wrapper = shallow(<ErrorBoundary />);
  wrapper.instance().componentDidCatch('something goes wrong...');
  expect(wrapper.instance().state.hasError).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});
