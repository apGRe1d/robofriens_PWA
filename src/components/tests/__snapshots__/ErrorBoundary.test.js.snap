// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when ErrorBoundary fires 1`] = `
<h1>
  Something went wrong..
</h1>
`;
